Computer History Museum Archive.js

```diff
@@ -0,0 +1,1000 @@
+	let items = {};
+	let found = false;
+	const entries = doc.querySelectorAll('p.objtext');
+
+	for (let i = 0; i < entries.length; i++) {
+		const titleDiv = entries[i].querySelector('span.objtitle a');
+		if (!titleDiv) continue;
+		const href = titleDiv.href;
+		const title = titleDiv.textContent;
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function scrape(doc, url) {
+	const fields = [...doc.querySelectorAll("div.field")];
+
+	let newItem = new Zotero.Item("book");
+	newItem.archive = 'Computer History Museum';
+
+	fields.forEach((nodeOriginal) => {
+		// clone the node to avoid altering the document.
+		const node = nodeOriginal.cloneNode(true)
+		const fieldTitleNode = node.querySelector("h4")
+		const fieldTitle = fieldTitleNode.textContent.trim()
+		switch(fieldTitle) {
+			case 'Title':
+				newItem.title = getContent(node, fieldTitle)
+				break;
+			case 'Date':
+				newItem.date = getContent(node, fieldTitle)
+				break;
+			case 'Participants':
+			case 'Contributor':
+				newItem.creators = getContributors(node)
+				break;
+			case 'Publisher':
+				newItem.publisher = getContent(node, fieldTitle)
+				break;
+			case 'Place of Publication':
+				newItem.publisher += '; ' + getContent(node, fieldTitle)
+				break;
+			case 'Extent':
+				newItem.numPages = getContent(node, fieldTitle).replace(' p.', '')
+				break;
+			case 'Lot Number':
+				newItem.archiveLocation = getContent(node, fieldTitle)
+				break;
+			case 'Description':
+				newItem.abstractNote = insertToTheStartOfAbstract(getContent(node, fieldTitle), newItem.abstractNote)
+				break;
+			case 'Biographical Notes':
+				newItem.abstractNote = insertToTheEndOfAbstract(getContent(node, fieldTitle), newItem.abstractNote, fieldTitle)
+				break
+			case 'Copyright Holder':
+				newItem.rights = getContent(node, fieldTitle)
+				break;
+			// the categories vary therefore many are collapsed under 'extra'.
+			case 'Type':
+			case 'Subject':
+			case 'Category':
+			case 'Collection Title':
+			case 'Copyright Holder':
+			case 'Dimensions':
+			case 'Credit':
+			case 'Place Manufactured':
+			case 'Manufacturer':
+			case 'System Requirements':
+			case 'Format':
+			case 'Series Title':
+			case 'Duration':
+			case 'Platform':
+			case 'Catalog Number':
+				newItem.extra = addToExtra(newItem.extra, getContent(node, fieldTitle), fieldTitle);
+				break
+			// parsing identifying numbers with existing Zotero fields e.g. callNumber & ISBN
+			case 'Identifying Numbers':
+				const numberDivs = node.querySelectorAll('tr')
+				numberDivs.forEach(div => {
+					const numberKey = div.querySelector('td.col1').textContent
+					const numberVal = div.querySelector('td.col2').textContent
+					switch(numberKey) {
+						case 'ISBN10':
+							newItem.ISBN = numberVal
+							break;
+						case 'LOC call num':
+							newItem.callNumber = numberVal
```

Comment:
`callNumber` refers to the call number of the item in its `libraryCatalog`, but in this case the library catalog is the CHM and the call number is for the LOC. This should go in Extra, if anywhere. I'm not sure we really need to keep it.

Reply:
Thanks! I moved it to extra. Considering that most users of the translator would probably like more archival location information than less especially when using Zotero for on-site research, I kindly ask to preserve this information.